lib/media/media_source_engine.js

```diff
@@ -452,9 +452,10 @@ shaka.media.MediaSourceEngine = class {
    * @param {boolean} sequenceMode
    */
   reinitText(mimeType, sequenceMode) {
-    if (!this.textEngine_) {
-      this.textEngine_ = new shaka.text.TextEngine(this.textDisplayer_);
+    if (this.textEngine_) {
+      this.textEngine_.destroy();
```

Comment:
This is necessary because otherwise the hot parser change doesn't work, I know it should work, but doing this works correctly... @joeyparrish, I need your advice here....

Test stream: https://www.fab-online.com/kgbzkjvhjfvf.m3u8

Reply:
Note: It seems that there is some error in the textengine when changing from a format that has an initialization segment to one that does not...